tests/types/test_schema.py

```diff
@@ -987,3 +997,353 @@ def test_infer_param_schema_with_errors():
+
+    with pytest.raises(
+        MlflowException,
+        match=r"Unsupported type 'invalid_type', expected instance of DataType, Array, Object or ",
+    ):
+        Property.from_json_dict(**{"p": {"type": "array", "items": "invalid_type"}})
+
+    with pytest.raises(
+        MlflowException,
+        match=r"Expected items to be a dictionary of Object JSON",
+    ):
+        Property.from_json_dict(**{"p": {"type": "array", "items": ("invalid_items_type",)}})
+
+    # test object
+    with pytest.raises(
+        MlflowException,
+        match=r"Missing keys in Object JSON. Expected to find keys `properties` and `type`",
+    ):
+        Property.from_json_dict(**{"p": {"type": "object"}})
+
+    with pytest.raises(
+        MlflowException, match=r"Expected properties to be a dictionary of Property JSON"
+    ):
+        Property.from_json_dict(**{"p": {"type": "object", "properties": "invalid_type"}})
+
+    with pytest.raises(
+        MlflowException, match=r"Expected properties to be a dictionary of Property JSON"
+    ):
+        Property.from_json_dict(
+            **{
+                "p": {
+                    "type": "object",
+                    "properties": {"p1": {"type": "string"}, "p2": "invalid_type"},
+                }
+            }
+        )
+
+
+def test_object_to_and_from_dict():
+    properties = []
+    dict_prop = {}
+    for data_type in DataType:
+        properties.append(Property(f"name_{data_type.name}", data_type))
+        dict_prop[f"name_{data_type.name}"] = {"type": data_type.name, "required": True}
+    obj = Object(properties)
+    assert obj.to_dict() == {
+        "type": "object",
+        "properties": dict(sorted(dict_prop.items())),
+    }
+    assert Object.from_json_dict(**json.loads(json.dumps(obj.to_dict()))) == obj
+
+
+def test_object_from_dict_with_errors():
+    with pytest.raises(
+        MlflowException,
+        match=r"Missing keys in Object JSON. Expected to find keys `properties` and `type`",
+    ):
+        Object.from_json_dict(**{"type": "object"})
+
+    with pytest.raises(
+        MlflowException, match=r"Type mismatch, Object expects `object` as the type"
+    ):
+        Object.from_json_dict(**{"type": "array", "properties": {}})
+
+    with pytest.raises(
+        MlflowException, match=r"Expected properties to be a dictionary of Property JSON"
+    ):
+        Object.from_json_dict(**{"type": "object", "properties": "invalid_type"})
+
+    with pytest.raises(
+        MlflowException, match=r"Expected properties to be a dictionary of Property JSON"
+    ):
+        Object.from_json_dict(
+            **{
+                "type": "object",
+                "properties": {"p1": {"type": "string"}, "p2": "invalid_type"},
+            }
+        )
+
+
+def test_array_to_and_from_dict():
+    for data_type in DataType:
+        arr = Array(data_type)
+        assert arr.to_dict() == {"type": "array", "items": data_type.name}
```

Comment:
Should the expected value look like `{"type": "array", "items": {"type": data_type.name}}`?

Reply:
I agree, this makes more sense. Let me update.